packages/aws-arch/scripts/generate-assets.ts

```diff
@@ -0,0 +1,392 @@
+import { GENERATED_DIR } from './common';
+import { listDirFiles, normalizeIdentifier } from "../src/utils";
+import { GENERAL_CATEGORY_ID, findAwsCategoryDefinition } from "../src/internal/categories/definitions";
+import { NORMALIZED_SERVICE_VARIANTS } from "../src/internal/services/variants";
+import { ASSET_DIRNAME, CATEGORY_ICON, EXT_PNG, EXT_SVG, SERVICE_ICON } from "../src/contants";
+
+export const ASSET_PACKAGE_ZIP_URL = 'https://d1.awsstatic.com/webteam/architecture-icons/q3-2022/Asset-Package_07312022.e9f969935ef6aa73b775f3a4cd8c67af2a4cf51e.zip';
+export const ASSET_PACKAGE_ZIP_FILENAME = path.basename(ASSET_PACKAGE_ZIP_URL);
+
+export const STATIC_ASSETS = path.resolve(__dirname, '..', 'static');
+export const ASSET_PACKAGE_DIR = path.resolve(__dirname, '..', ASSET_DIRNAME);
+
+export const AWS_ASSETS_TS = path.join(GENERATED_DIR, 'assets.ts');
+
+export const ASSETS_MARKDOWN = path.join(__dirname, "..", 'ASSETS.md');
+
+export const ASSET_PATTERNS = {
+  CATEGORY: /Category-Icons_\d+\/Arch-Category_64\/Arch-Category_(?<category>[^_]+)_64\.svg$/,
+  SERVICE: /Architecture-Service-Icons_\d+\/Arch_(?<category>[^/]+)\/(Arch_)?64\/Arch_(?<serviceName>[^_]+)_64\.svg$/,
+  RESOURCE: /Resource-Icons_\d+\/Res_(?<category>[^/]+)\/Res_48_(Light|Dark)\/Res_(?<serviceName>[^_]+)_(?<resourceName>[^_ ]+) ?_48_((?<theme>Light|Dark))\.svg$/,
+  GENERAL: /(Arch|Res)_General-Icons\/(64|Res_48_(Light|Dark))\/(Arch|Res)_(?<generalIcon>[^_]+)_.*(?<theme>Light|Dark).*\.svg$/,
+};
+
+const EC2_INSTANCE_PATTERN = /compute\/ec2\/(?<instanceType>\w+)_instance$/i;
+
+const IOT_THING_PATTERN = /internet_of_things\/iot_thing\/(?<thingType>\w+)$/i;
+
+export const ASSET_SIZE = 450;
+
+const DEFAULT_THEME = 'light';
+
+// List of "general" resources used by other resources from common resources
+// Lots of services have logs, rules, policies, which are pulled from IAM
+// TODO: convert generic assets to grayscale
+const GENERAL_ALIASES = {
+  alarm: 'management_governance/cloudwatch/alarm',
+  attachment: 'networking_content_delivery/transit_gateway/attachment',
+  config: 'general/gear',
+  configuration: 'general/gear',
+  credentials_long: 'security_identity_compliance/iam/long_term_security_credential',
+  credentials_temp: 'security_identity_compliance/iam/temporary_security_credential',
+  encrypted_data: 'security_identity_compliance/iam/encrypted_data',
+  event: 'management_governance/cloudwatch/event_event_based',
+  interactive_video: 'general/multimedia',
+  key: 'security_identity_compliance/iam/permissions',
+  logs: 'management_governance/cloudwatch/logs',
+  policy: 'security_identity_compliance/iam/add_on',
+  role: 'security_identity_compliance/iam/role',
+  rule: 'management_governance/cloudwatch/rule',
+  scheduled_event: 'management_governance/cloudwatch/event_time_based',
+  template: 'management_governance/cloudformation/template',
+ } as const;
+
+export async function generate () {
```

Comment:
Is there a way to split this function up into multiple functions as its pretty hefty atm?

Reply:
The local variables are reused throughout, so while splitting it is definitely possible not sure it would become more readable nor maintainable at this point. Before this package become "stable" will definitely look into splitting it out, but not sure would get value for effort at this state. Happy to discuss and align if still a concern.